docs/content/preview/explore/ysql-language-features/join-strategies.md

```diff
@@ -0,0 +1,256 @@
+               Storage Table Read Execution Time: 1001.211 ms
+ Planning Time: 0.231 ms
+ Execution Time: 2002.540 ms
+ Storage Read Requests: 2
+ Storage Read Execution Time: 2002.137 ms
+ Storage Execution Time: 2002.137 ms
+ Peak Memory Usage: 141 kB
+```
+
+## Batched Nested Loop Join (BNL)
+
+In the case of the Nested loop joins, the inner table is accessed multiple times one for each outer table row. This would lead to multiple RPC requests across the different nodes in the cluster, making this join strategy very slow as the outer table gets larger.
+
+To reduce the number of requests sent across the nodes during the Nested Loop join, YugabyteDB adds a new optimization to batch multiple keys of the outer table into one RPC request. This batch size can be controlled using the GUC variable `yb_bnl_batch_size`, which defaults to `1` (which effectively means that the feature is `OFF`). The suggested value for this variable is `1024`.
+
+If `yb_bnl_batch_size` is greater than `1`, the optimizer will try to adopt the batching optimization when other join strategies are not fit for the current query.
+
+To fetch all scores of students named `Natasha` who have scored more than `70` in any subject using Batched nested loop join, you would execute,
+
+```sql
+explain (analyze, dist, costs off)
+/*+
+    nestloop(students scores)
+    set(yb_bnl_batch_size 1024)
+    set(yb_enable_optimizer_statistics on)
+*/
+SELECT name, subject, score 
+      FROM students JOIN scores USING(id) 
+      WHERE name = 'Natasha' and score > 70;
+```
+
+And the query plan would be similar to:
+
+```sql
+ YB Batched Nested Loop Join (actual time=4004.255..4004.297 rows=18 loops=1)
+   Join Filter: (students.id = scores.id)
+   ->  Index Scan using idx_name on students (actual time=2001.861..2001.870 rows=8 loops=1)
+         Index Cond: ((name)::text = 'Natasha'::text)
+         Storage Table Read Requests: 1
+         Storage Table Read Execution Time: 1001.078 ms
+         Storage Index Read Requests: 1
+         Storage Index Read Execution Time: 1000.589 ms
+   ->  Index Scan using idx_id on scores (actual time=2002.290..2002.305 rows=18 loops=1)
+         Index Cond: (id = ANY (ARRAY[students.id, $1, $2, ..., $1023]))
+         Remote Filter: (score > 70)
+         Storage Table Read Requests: 1
+         Storage Table Read Execution Time: 1001.356 ms
+         Storage Index Read Requests: 1
+         Storage Index Read Execution Time: 1000.670 ms
+ Planning Time: 0.631 ms
+ Execution Time: 4004.541 ms
+ Storage Read Requests: 4
+ Storage Read Execution Time: 4003.693 ms
+ Storage Execution Time: 4003.693 ms
+ Peak Memory Usage: 476 kB
+```
+
+The key point to note here is index condition: `(id = ANY (ARRAY[students.id, $1, $2, ..., $1023]))` which led to `loops=1` as the lookup was batched. In the case of [nested loop join](#nested-loop-join), this was `loops=8`. This batching would drastically improve performance in many scenarios.
```

Comment:
*the index condition

Reply:
> which led to `loops=1` as the...

This part of the sentence reads kind of confusing. Maybe if you could specify that you are talking about the inner scan here that could make it read clearer. `loops=1` can be found on the outer scan as well.

> this was `loops=8`
I think this would read easier if "this" was avoided. Maybe something like 
"In the case of ..., the inner side looped 8 times as seen by....".

>This batching would drastically improve performance in many scenarios.
"would" reads in the past tense to me here 
Maybe something like,

Such batching can drastically improve ...
Such batching drastically improves...

Reply:
fixed